### PR TITLE
Support dotnet pack on libs src project

### LINKED ISSUE #37960: Support dotnet pack on libraries projects
`dotnet pack` currently does nothing when invoked on an library project. We should be able to route `dotnet pack` invoked on the source project to the pkgproj and invoke its build target.

In addition to that it would be nice if the pkgproj build output doesn't point to the `nuspec` but to the `nupkg` instead (same what NuGet does). //update: already done by @ericstj: https://github.com/dotnet/arcade/commit/b0b76b68f059c3f94aba2818ae0da601adcc7096

Long term, we should consider getting rid of `pkgproj`s and enable `dotnet pack` natively on our source projects. This probably involves extending the existing NuGet infrastructure, presumably with a new msbuild sdk similar to the existing [Microsoft.DotNet.Build.Tasks.Packaging](https://github.com/dotnet/arcade/tree/master/src/Microsoft.DotNet.Build.Tasks.Packaging) infra package. Related: https://github.com/dotnet/arcade/issues/1033

See https://github.com/dotnet/runtime/issues/37961 for exposing a single entry point for generating the library package.

cc @joperezr @ericstj @Anipik

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/37960

Not adding the redirection to the ref project to make the src project the single entrypoint for `dotnet pack`. This probably needs some documentation updates as well.

### CODE REVIEW

File: eng/packaging.targets [hunk 1]
```diff
@@ -26,4 +26,16 @@
                       '$(_excludeCompile)' == 'true' and
                       '%(Dependency.Identity)' != '_._'" />
   </Target>
+
+  <PropertyGroup>
+    <PkgProjPath>$(MSBuildProjectDirectory)\..\pkg\$(MSBuildProjectName).pkgproj</PkgProjPath>
+    <!-- Point to our cusotm pack target which builds the pkgproj instead. -->
```
Comment (safern) [inline eng/packaging.targets:32]:
```suggestion
    <!-- Point to our custom pack target which builds the pkgproj instead. -->
```

**Review by safern — APPROVED** [review]
With this change, if I do dotnet pack in a src proj, will it restore the ref and src for me?

File: eng/packaging.targets [hunk 2]
```diff
@@ -26,4 +26,16 @@
...
+    <PackDependsOn Condition="'$(IsSourceProject)' == 'true' and
+                              Exists('$(PkgProjPath)')">_BuildPkgProj</PackDependsOn>
```
Comment (ericstj) [inline eng/packaging.targets:34]: Cool.  What does Pack target do for our csprojs. Is it a noop?

**Review by ericstj — APPROVED** [review]

File: eng/packaging.targets [hunk 3]
```diff
@@ -26,4 +26,16 @@
...
+  </PropertyGroup>
+
+  <Target Name="_BuildPkgProj">
+    <MSBuild Projects="$(PkgProjPath)"
+             Targets="Build" />
```
Comment (ericstj) [inline eng/packaging.targets:39]: Nit: you can omit Targets.

**Review by ericstj — APPROVED** [review]

**Comment by ViktorHofer** [comment]
> With this change, if I do dotnet pack in a src proj, will it restore the ref and src for me?

Yes, it will incrementally restore and build both the src project and if there's a P2P to the ref project as well, then also that one.

File: eng/packaging.targets (hunk 3 above)
Comment (ViktorHofer) [reply eng/packaging.targets:39]: I intentionally put it there as I wasn't sure what the default target for a pkgproj is. If there isn't one specified, MSBuild always picks the first target found.

File: eng/packaging.targets (hunk 2 above)
Comment (ViktorHofer) [reply eng/packaging.targets:34]: Yeah because `IsPackable` is currently set to false.

**Pull request merged**